tools/slang-test/render-api-util.h

```diff
@@ -0,0 +1,61 @@
+#ifndef SLANG_RENDER_API_UTIL_H
+#define SLANG_RENDER_API_UTIL_H
+
+#include "../../source/core/slang-string.h"
+#include "../../source/core/slang-result.h"
+
+
+enum class RenderApiType
+{
+    kUnknown = -1,
+    kOpenGl = 0,
+    kVulkan,
+    kD3D12,
+    kD3D11,
+    kCountOf,
```

Comment:
Not sure I'm a fan of the `Of` suffix on `kCountOf` and `kAllOf`, but that's a bikeshed sort of thing.

One nice thing about the non-`enum class` approach with `kEnumType_TagName` naming is that the `kEnumTypeCount` case is grammatically reasonable, and is also visually distinguished from the other cases so that it wouldn't be too confusing.

Reply:
Ok my bad - I thought that was the convention that was being used, but you are just using it for 'constants'. The carbonite team were doing this. They just moved to use 'e' prefixes. I'm not really mad about 'k' or 'e' prefixes for enums personally. Will fix.